### PR TITLE
Add some comments for background.js

### PR DESCRIPTION
**Pull request by austinkregel**

### CODE REVIEW

**Comment by austinkregel** [comment]
I added a few tweaks... :)

The main ones are I "fixed" the variables that were declared like python variables to javascript variables, and I also added documentation.

Then after everything was "converted" I added a promised base HTTP request system for dat syntax sugar

    request('get', 'url')
        .then(function() {})
        .catch(function() {});

File: extension/background.js [hunk 1]
```diff
@@ -1,79 +1,107 @@
...
             handle_contrib_data(dom);
-        }
-    };
-    xhr.open("GET", contrib_url, true);
-    xhr.send();
+        });
 }
 
+/**
+ * Make AJax request and call closure when ready.
```
Comment (daspecster) [inline extension/background.js]: `/AJax/AJAX or Ajax or ajax`?

File: extension/background.js [hunk 2]
```diff
@@ -1,79 +1,107 @@
...
-    imageData: context.getImageData(0, 0, 19, 19)
-  });
+    chrome.browserAction.setIcon({
+        imageData: context.getImageData(0, 0, 19, 19)
+    });
 }
 
+/**
+ * Handle the contributions data by parsing through the temporary dom
+ * Then calls draw to build the icon for the extension icon
```
Comment (daspecster) [inline extension/background.js]: Maybe add a `.` at the end of the sentences like the ones above.

File: extension/background.js [hunk 3]
```diff
@@ -1,79 +1,107 @@
...
+    var last_day = days[days.length - 1];
 
     var todays_contributions = last_day.getAttribute("data-count");
     var color = last_day.getAttribute("fill");
     draw(color, todays_contributions);
 }
 
-
+/**
+ * Get the contributions for the given user and parse the data
```
Comment (daspecster) [inline extension/background.js]: ditto `.`

File: extension/background.js [hunk 4]
```diff
@@ -1,79 +1,107 @@
...
+                reject(xhr);
+            }
+        };
+        xhr.onerror = function () {
+            reject(xhr);
+        };
+        xhr.send();
+    });
+}
+// Grab the username from github
```
Comment (daspecster) [inline extension/background.js]: ditto `.`, just since the one below this has it.

File: extension/background.js [hunk 5]
```diff
@@ -1,79 +1,107 @@
-username = "";
+var username = "";
```
Comment (daspecster) [inline extension/background.js:1]: Does this actually change the scope in this case? I think your addition is more correct but I can't remember if JS attaches the `username` on to `window` or if `var username` attaches the variable to a more local scope?

**Review by daspecster — CHANGES_REQUESTED** [review]
Looks awesome man! Thanks!

File: extension/background.js (hunk 5 above)
Comment (austinkregel) [reply extension/background.js:1]: Either way your whole program will  have access to it. If it needs to be on the window than I can change it to `window.username = "";` instead if you'd like

File: extension/background.js (hunk 1 above)
Comment (austinkregel) [reply extension/background.js]: Honestly, I as going for AJAX but looks like my shift key gave out. :/ Do you have any preferences?

Comment (daspecster) [reply extension/background.js]: I concur doctor...I concur. `AJAX`.

File: extension/background.js (hunk 5 above)
Comment (daspecster) [reply extension/background.js:1]: As it is now is perfect.

**Review by daspecster — APPROVED** [review]
Looks good!

**Pull request merged**